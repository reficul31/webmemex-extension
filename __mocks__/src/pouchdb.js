import PouchDB from 'pouchdb-node'
import PouchDBMemory from 'pouchdb-adapter-memory'

PouchDB.plugin(PouchDBMemory)

const pouchdbOptions = {
    name: 'testdb',
    auto_compaction: true,
    adapter: 'memory',
}

const db = PouchDB(pouchdbOptions)
export default db
