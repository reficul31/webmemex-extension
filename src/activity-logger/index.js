// Stuff that is to be accessible from other modules (folders)

import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const visitKeyPrefix = 'visit/'

/**
* Creates an _id string given the variables, or vice versa parses such strings
* We simply use the creation time for the id, for easy chronological sorting.
* We add a random string we call a 'nonce' to prevent accidental collisions.
*
* @param {Number} timestamp - timestamp of the document
* @param {Number} nonce - random 10 digit number
*
* @returns {String} - string id of the document
*/
export const convertVisitDocId = docuri.route(`${visitKeyPrefix}:timestamp/:nonce`)

/**
* Extract the type, timestamp and nonce from an ID of a document
*
* @param {String} - string id of the document
*/
const convertAnyTimestampedDocId = docuri.route(':type/:timestamp/:nonce')
export const getTimestamp = doc =>
    Number.parseInt(convertAnyTimestampedDocId(doc._id).timestamp)

/**
* Generate a visit document ID from the timestamp and a random string
*
* @param {Number} timestamp - timestamp of the visit document
* @param {Number} nonce - random 10 digit number
*
* @returns {string} - document id of the visit document
*/
export function generateVisitDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertVisitDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}

/**
* Tell whether a page can be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page is loggable
*/
export function isLoggable({url}) {
    // Only http(s) pages. Ignoring data URLs, newtab, ...
    const loggableUrlPattern = /^https?:\/\//
    return loggableUrlPattern.test(url)
}

/**
* Tell whether a page should be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page should be stored
*/
export function shouldBeLogged({url}) {
    // Currently, we log everything that we think we can log.
    // TODO Add ignore-/blacklist lookup here (issue #22, #94)
    return isLoggable({url})
}
