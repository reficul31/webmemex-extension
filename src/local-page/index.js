import get from 'lodash/fp/get'

/**
* Extract the local page URL from the page instance.
*
* @param {Page} page - stored page in the database
*
* @returns {string} - URL used to refer to the local page
*/
export function hrefForLocalPage({page}) {
    // Check if it has a stored page attached at all.
    if (!get(['_attachments', 'frozen-page.html'])(page)) {
        return undefined
    }
    const pageId = page._id
    const url = `/local-page.html?page=${encodeURIComponent(pageId)}`
    const hash = (page.url && page.url.split('#')[1])
    const href = (hash !== undefined) ? url + '#' + hash : url
    return href
}
