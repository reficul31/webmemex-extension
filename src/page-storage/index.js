import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const pageKeyPrefix = 'page/'

export const convertPageDocId = docuri.route(`${pageKeyPrefix}:timestamp/:nonce`)

/**
* Generate a page doc id
*
* @param {Number} timestamp - timestamp of the page
* @param {string} nonce - random string of 10 digit number
*
* @returns {string} - doc id of the page as a string
*/
export function generatePageDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertPageDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}
