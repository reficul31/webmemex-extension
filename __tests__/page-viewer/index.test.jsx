import React from 'react'
import { LinkToLocalVersion, localVersionAvailable } from 'src/page-viewer'
import renderer from 'react-test-renderer'
import page from './page.data.json'

describe('LinkToLocalVersion snapshot tests', () => {
    test('should test against the given snapshot', () => {
        const component = renderer.create(
            <LinkToLocalVersion page={page}>
                <img src='img/save-icon.png' alt='ðŸ’¾ saved' />
            </LinkToLocalVersion>
            )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('localVersionAvailable tests', () => {
    test('should return true when the page is available', () => {
        let page = {
            '_attachments': {
                'frozen-page.html': 'htmldoc',
            },
        }
        expect(localVersionAvailable({page})).toBeTruthy()
    })

    test('should return false when the content is not available', () => {
        let page = {
            '_attachments': {},
        }
        expect(localVersionAvailable({page})).toBeFalsy()
    })
})
