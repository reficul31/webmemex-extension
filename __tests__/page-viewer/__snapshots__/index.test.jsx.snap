// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToLocalVersion snapshot tests should test against the given snapshot 1`] = `
<a
  href="/page-viewer/localpage.html?page=page/1495120848122/1895257124"
  title="Stored version available"
>
  <img
    alt="ðŸ’¾ saved"
    src="img/save-icon.png"
  />
</a>
`;
